packages/babel-plugin-transform-destructuring/test/fixtures/destructuring/for-expressions-nest/input.js

```diff
@@ -0,0 +1 @@
+for ( let x in a ) for ( var { } = x ; ; ) ;
```

Comment:
Can you add a new test case? So that we cover the branch `inForInit === false` but `node` is an `ExpressionStatement`.

```js
for ( let x in a ) for ( ; ; ) var { } = x ;
```